test/contrib/sam/test_model.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+import torch
+
+from kornia.contrib.sam import Sam, SamConfig
+from kornia.core import pad
+from kornia.testing import BaseTester
+
+
+def _pad_rb(x, size):
+    """Pads right bottom."""
+    pad_h = size - x.shape[-2]
+    pad_w = size - x.shape[-1]
+    return pad(x, (0, pad_w, 0, pad_h))
+
+
+class TestSam(BaseTester):
+    def test_smoke(self):
+        ...
+
+    @pytest.mark.parametrize('batch_size', [1, 3])
+    @pytest.mark.parametrize('N', [2, 5])
+    @pytest.mark.parametrize('multimask_output', [True, False])
+    def test_cardinality(self, device, batch_size, N, multimask_output):
+        # SAM: don't supports float64
```

Comment:
Why not?

Reply:
good question, need to deep review, but almost all tests fail on some internal part complaining that he wanted a float and receive a double